### PR TITLE
Add an insert and update test

### PR DESCRIPTION
**Pull request by GTZL1**

### CODE REVIEW

File: src/main/java/ch/heigvd/pro/b04/polls/ServerPollIdentifier.java [hunk 1]
```diff
@@ -24,6 +25,7 @@
   private long idPoll;
 
   @ManyToOne
+  @Setter
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/polls/ServerPollIdentifier.java]: This is not needed - you could use the `Builder` to do exactly the same thing.

File: src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java [hunk 2]
```diff
@@ -139,10 +141,75 @@ public void testParticipantCannotInsertOrUpdateOrDeleteQuestion() {
...
+            .build())
+        .title("Do you love the Frostclaws ?")
+        .build();
+
+    ServerPoll pollTemp2 = ServerPoll.builder()
+        .idPoll(ServerPollIdentifier.builder().idPoll(223).build())
+        .pollServerQuestions(Set.of(q2))
+        .build();
+
+    q2.getIdServerQuestion().setIdxPoll(pollTemp2);
```
Comment (alexandrepiveteau) [inline src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java:220]: Please use the `Builder` (and reorder code if necessary).

File: src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java [hunk 3]
```diff
@@ -139,10 +141,75 @@ public void testParticipantCannotInsertOrUpdateOrDeleteQuestion() {
...
     lenient().when(participantRepository.findByToken("t1")).thenReturn(Optional.of(aloy));
-    //lenient().when(repo.delete(Mockito.any())).thenReturn(new ServerMessage("success"));
 
     assertThrows(WrongCredentialsException.class, () -> cc.insert(1, 123, "t1", c2));
     assertThrows(WrongCredentialsException.class, () -> cc.updateQuestion(1, 123, 1, "t1", c2));
     assertThrows(WrongCredentialsException.class, () -> cc.deleteQuestion(1, 123, 1, "t1"));
   }
+
+  @Test
+  public void testModeratorCanInsertOrUpdateQuestionOnlyWithRightAccess()
```
Comment (alexandrepiveteau) [inline src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java]: Unit tests should test a single thing at a time (especially if they're independent). Could you please separate this method in two ?

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java (hunk 2 above)
Comment (GTZL1) [reply src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java:220]: We can't reorganize code here, that's what I said on Telegram: using exclusively the Builders, the snake bites his tail. We have to set idxPoll after.

File: src/main/java/ch/heigvd/pro/b04/polls/ServerPollIdentifier.java (hunk 1 above)
Comment (GTZL1) [reply src/main/java/ch/heigvd/pro/b04/polls/ServerPollIdentifier.java]: I'll modify it

Comment (GTZL1) [reply src/main/java/ch/heigvd/pro/b04/polls/ServerPollIdentifier.java]: You mean to rebuild it entierly ?

File: src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java (hunk 3 above)
Comment (GTZL1) [reply src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java]: it's done :)

File: src/main/java/ch/heigvd/pro/b04/polls/ServerPollIdentifier.java (hunk 1 above)
Comment (alexandrepiveteau) [reply src/main/java/ch/heigvd/pro/b04/polls/ServerPollIdentifier.java]: This class has a Builder constructor. Why not use it ?

**Review by alexandrepiveteau — APPROVED** [review]

**Pull request merged**